v4/export/writer.go

```diff
@@ -137,12 +141,19 @@ type csvOption struct {
 	delimiter []byte
 }
 
-func newOutputFileNamer(ir TableDataIR) *outputFileNamer {
-	return &outputFileNamer{
-		Index: ir.ChunkIndex(),
+func newOutputFileNamer(ir TableDataIR, bothEnabled bool) *outputFileNamer {
+	o := &outputFileNamer{
 		DB:    ir.DatabaseName(),
 		Table: ir.TableName(),
 	}
+	if bothEnabled {
+		o.id = 0
+		o.idFormat = fmt.Sprintf("%09d", ir.ChunkIndex()) + "%04d"
```

Comment:
13 digits will overflow an int32. Do tools that read these files parse out the number, or just rely on it for filesystem ordering?

Reply:
so far all of myloader, Lightning and DM only rely on filesystem order, and won't parse the number. Lightning is the only one requiring that part to be `[0-9]+` because the regex was written too tightly.